shared-module/qrio/QRDecoder.c

```diff
@@ -138,14 +138,30 @@ mp_obj_t shared_module_qrio_qrdecoder_decode(qrdecoder_qrdecoder_obj_t *self, co
     mp_obj_t result = mp_obj_new_list(0, NULL);
     for (int i = 0; i < count; i++) {
         quirc_extract(self->quirc, i, &self->code);
-        if (quirc_decode(&self->code, &self->data) != QUIRC_SUCCESS) {
-            continue;
+        mp_obj_t code_obj;
+        if (decode) {
```

Comment:
This is a bit weird to me. Almost always there is a 1:1 relationship between Python functions and their C common_hal parts. Could you refactor this so that `find()` and `decode()` share code but have two separate common_hal functions that each include their piece of this branch?

Reply:
The only way I can think of doing this cleanly is to rename this function, and then have two one-liner common_hal functions that call it, with true or false. Would that be better?

Reply:
That's a little better but not what I'm thinking. Ideally this code would be split apart. I bet you can factor out the start through quirc_end and return quirc_count(). You'll end up with a loop in each function that mirrors this one (but simpler without the if).

Reply:
I would still have two functions identical except for what is inside that if, so I don't see much point? I will try to come up with something.

Reply:
I moved as much as I could without passing too many variables around to a common function, but there is still a lot of duplication. I'm not happy with it, but I can't see how to improve it.